src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -447,7 +448,10 @@ private function getModulesOnDisk($skip_main_class_attributes = false)
             ->ignoreVCS(true);
 
         foreach ($modulesDirsList as $moduleDir) {
-            $moduleName = $moduleDir->getFilename();
+            $php_file_path = _PS_MODULE_DIR_.$moduleDir->getFilename().'/'.$moduleDir->getFilename().'.php';
+            if (file_exists($php_file_path)) {
+                $moduleName = $moduleDir->getFilename();
```

Comment:
Maybe you should always declare the variable `$moduleName` and bypass the module if the file does not exist. Leaving the variable undefined will throw an exception later (L.456), which is not really good in my opinion.
It would be better if we handle it properly with a `continue;`, because it's useless to even try to load it if the file does not exist.

We can have something like this:
```php
$moduleName = $moduleDir->getFilename();
if (!file_exists(_PS_MODULE_DIR_.$moduleName.'/'.$moduleName.'.php')) {
    continue;
}
```

What do you think ?